### ISSUE TITLE
Updating dateFormat with datePickerConfig causes error in the console

### ISSUE THREAD

**Issue by aninde** [handsontable/handsontable#7282]
### Description
Updating dateFormat with datePickerConfig causes error in the console.

Probably its caused by the fragment of the config of DatePicker `showWeekNumber` (regardless of whether true or false). After updateSettings and opening the editor on the calendar cell, an error appears in the console on line 70787 of `handsontable.full.js` `v8.0.0`.

```
for (var c = 0; c < opts.numberOfMonths; c++) {
                html += '<div class="pika-lendar">' + renderTitle(this, c, this.calendars[c].year, this.calendars[c].month, this.calendars[0].year, randId) + this.render(this.calendars[c].year, this.calendars[c].month, randId) + '</div>';
            }
```
There is an issue on the Pikaday issue board related to `showWeekNumber` https://github.com/Pikaday/Pikaday/issues/878.

![bug-ff](https://user-images.githubusercontent.com/30186818/93483534-33595f00-f901-11ea-92e4-c7ac98d4819b.gif)

### Steps to reproduce

1. Open demo with browser console.
2. Update settings by clicking button.
3. Open editor at any cell.

### Demo

https://jsfiddle.net/aninde/pvos2c4r/ - v8.0.0
https://jsfiddle.net/aninde/hy5ckotz/ - v2.0.0

### Your environment
* Handsontable version: 2+
* Browser Name and version: not related
* Operating System: macOS Catalina 10.15.6

**Comment by scarletfog** [comment]
I just stumbled upon this issue. Isn't it caused by the fact that the settings are indeed updated with a new format but the data remains the same? It sounds quite usual that "01/14/2017" is no longer properly recognized as a date 🤔

**Comment by rikkert** [comment]
There is a new version of pikaday, 1.8.2 that fixed some week number issue. 
Is it still a problem?

**Comment by AMBudnik** [comment]
Thank you for sharing the good news @rikkert! 
I see that the issue @aninde referenced is also fixed by Pikaday 1.8.2. 
We need to test it out.
